File: 스택/뒤에 있는 큰 수 찾기.java
import java.util.*;

class Solution {
    public int[] solution(int[] numbers) {
        int[] answer = new int[numbers.length];
        Stack<Integer> s = new Stack<>();
        for(int i=0; i<numbers.length; i++){
            while(!s.empty() && numbers[s.peek()]<numbers[i]){
                answer[s.peek()] = numbers[i];
                s.pop();
            }
            s.push(i);
        }
        while(!s.empty()){
            answer[s.peek()] = -1;
            s.pop();
        }
        return answer;
    }
}

File: 스택/뒤에 있는 큰 수 찾기.cpp
#include <stack>
#include <vector>

using namespace std;

vector<int> solution(vector<int> numbers) {
    vector<int> answer(numbers.size(),0);
    stack<int> s;
    for(int i=0; i<numbers.size(); i++){
        while(!s.empty() && numbers[s.top()] < numbers[i]){
            answer[s.top()] = numbers[i];
            s.pop();
        }
        s.push(i);
    }
    while(!s.empty()){
        answer[s.top()] = -1;
        s.pop();
    }
    return answer;
}
